knowledge_graphs/hardware_optimized_config.py

```
@@ -0,0 +1,361 @@
+
+@dataclass
+class HardwareProfile:
+    """Hardware profile for optimization."""
+    cpu_cores: int
+    cpu_threads: int
+    total_ram_gb: int
+    available_ram_gb: int
+    gpu_vram_gb: int
+    cpu_model: str
+    
+    def __post_init__(self):
```

Ruff (0.12.2)

Missing return type annotation for special method `__post_init__`

(ANN204)